cli/cli.go

```diff
@@ -572,23 +570,50 @@ Not after (expires): %s (%s)
 }
 
 func userHomeDir() string {
-	if runtime.GOOS == "windows" {
-		home := os.Getenv("HOMEDRIVE") + os.Getenv("HOMEPATH")
-		if home == "" {
-			home = os.Getenv("USERPROFILE")
-		}
-		return home
+	home, err := os.UserHomeDir()
+	if err != nil {
+		panic("HOME directoy is not defined")
 	}
-	return os.Getenv("HOME")
+	return home
 }
 
 func getConfigDir(appName string) string {
 	configDirEnv := strings.ToUpper(appName) + "_CONFIG_DIR"
-
 	configDir := os.Getenv(configDirEnv)
 
 	if configDir == "" {
-		configDir = path.Join(userHomeDir(), "."+appName)
+		// Create new config directory
+		configBase, _ := os.UserConfigDir()
+		configDir = filepath.Join(configBase, appName)
+
+		// Check for legacy config dir
+		legacyConfigDir := filepath.Join(viper.GetString("home-directory"), "."+appName)
+		if _, err := os.Stat(legacyConfigDir); err == nil {
+			// Only migrate if the new config dir doesn't exist, so that this
+			// is a one-time operation. There are edge cases where configs could
+			// get lost if we migrate every time (e.g. running an old version
+			// that creates an empty ~/.restish/apis.json).
+			if _, err := os.Stat(configDir); err != nil {
+				// Define files to migrate
+				for _, filename := range []string{
+					"config.json",
+					"apis.json",
+					"cache.json",
```

Comment:
Is that trailing comma going to be a problem?

Reply:
Nope, in fact Go (fmt or vet or one of those tools) will complain if it's missing.